lib/arm/cpu_features.c

```diff
@@ -167,6 +167,56 @@ static u32 query_arm_cpu_features(void)
 	}
 	return features;
 }
+
+#elif defined(_WIN32)
+
+/*
+ * https://docs.microsoft.com/en-us/cpp/build/arm64-windows-abi-conventions
+ * "The ARM64 version of Windows presupposes that it's running on an ARMv8 or
+ * later architecture at all times. Both floating-point and NEON support are
+ * presumed to be present in hardware."
+ * Some (but not all) other features are exposed in various Windows APIs.
+ * However, the cached values of the CP15 regs can also be read out of the
+ * registry.
+ */
+
+#include <Windows.h>
+#include <stdio.h>
+
+/* Note this reads cpu 0 only */
+static bool read_cp_reg(u64* value, u32 reg)
+{
+	DWORD value_len = sizeof(*value);
+	char value_name[12];
+	if (snprintf(value_name, sizeof(value_name), "CP %x", reg) < 0)
```

Comment:
Does this actually work?  Are you sure it doesn't need to be `CP %08x` or `CP 0x%08x`?

Also, this is the wrong way to check the return value of `snprintf()`.

Reply:
snprintf check updated.

the change is based on https://github.com/shuffle2/libdeflate/commits/msvc, so I cannot tell/validate/test it

similar code from msft: https://github.com/microsoft/SymCrypt/blob/master/lib/cpuid.c

Reply:
I don't see anything that queries the registry in that code.

Anyway, I'm not going to merge something that no one has tested...

Reply:
code came from https://github.com/shuffle2/libdeflate/commits/msvc
@shuffle2, did you test it on arm?